### PR TITLE
Adding assertion to check for regular JSON inputs of size greater than `INT_MAX` bytes

### PR DESCRIPTION
**Pull request by shrshi**
## Description
Addresses #17017 

Libcudf does not support parsing regular JSON inputs of size greater than `INT_MAX` bytes. Note that the batched reader can only be used for JSON lines inputs.

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [X] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/src/io/json/read_json.cu [hunk 1]
```diff
@@ -351,10 +351,17 @@ table_with_metadata read_json(host_span<std::unique_ptr<datasource>> sources,
...
-  std::size_t chunk_offset            = reader_opts.get_byte_range_offset();
-  std::size_t chunk_size              = reader_opts.get_byte_range_size();
-  chunk_size                          = !chunk_size ? total_source_size - chunk_offset
-                                                    : std::min(chunk_size, total_source_size - chunk_offset);
+
+  // Batching is enabled only for JSONL inputs, not regular JSON files
+  if (!reader_opts.is_enabled_lines()) {
+    CUDF_EXPECTS(total_source_size < std::numeric_limits<int32_t>::max(),
+                 "Parsing Regular JSON inputs of size greater than INT_MAX bytes is not supported");
+  }
```
Comment (mhaseeb123) [inline cpp/src/io/json/read_json.cu:359]:
```suggestion
  CUDF_EXPECTS(reader_opts.is_enabled_lines() or total_source_size < std::numeric_limits<int32_t>::max(),
               "Parsing Regular JSON inputs of size greater than INT_MAX bytes is not supported");
```

**Review by mhaseeb123 — APPROVED** [review]
Minor comment, may need style fix. LGTM otherwise!

**Review by karthikeyann — COMMENTED** [review]
does `check_input_size` need an update as well ? because it uses 
`std::numeric_limits<cudf::io::json::SymbolOffsetT>::max()` aka `uint32_t`

**Comment by shrshi** [comment]
> does `check_input_size` need an update as well ? because it uses `std::numeric_limits<cudf::io::json::SymbolOffsetT>::max()` aka `uint32_t`

That's correct, I've updated `check_input_size` to make it consistent with the check in `read_json`.

**Review by vuule — APPROVED** [review]

File: cpp/src/io/json/read_json.cu (hunk 1 above)
Comment (shrshi) [reply cpp/src/io/json/read_json.cu]: Addressed!

**Review by karthikeyann — APPROVED** [review]

**Pull request merged**